libraries/joomla/form/fields/sql.php

```
@@ -284,7 +284,19 @@ protected function getOptions()
 		}
 		catch (Exception $e)
```

Please change this catch to `catch (JDatabaseExceptionExecuting $e)`

There is no reason for it to catch all Exceptions.